archive/tarfile.go

```diff
@@ -0,0 +1,67 @@
+package archive
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	V "github.com/gspivey/drayage/volume"
+)
+
+//WriteToTar tars any found volumes
+func WriteToTar() (err error) {
+	volumes, err := V.ListVolumes()
+	for _, vol := range volumes {
+		dir, err := os.Open(vol.Mountpoint)
+		if err != nil {
+			return fmt.Errorf("can't open %v (did you use sudo?)", vol.Name)
+		}
+		defer dir.Close()
+
+		files, err := dir.Readdir(0)
+		if err != nil {
+			return fmt.Errorf("can't read directory")
+		}
+
+		destfile := vol.Name + ".tgz"
+
+		//
+		tarfile, err := os.Create(destfile)
+		defer tarfile.Close()
+
+		var fileWriter io.WriteCloser = tarfile
```

Comment:
Nothing to address: Fascinating. You passed each type of iowriter is acting as a "filter".cool